.github/workflows/slash-command-merge.yml

```diff
@@ -51,8 +51,50 @@ jobs:
               return;
             }
 
+
+            function getSegments(rex, str) {
+              var segments = [];
+              var lastIndex = 0;
+              var match;
+              rex.lastIndex = 0; // In case there's a dangling previous search
+              while (match = rex.exec(str)) {
+                if (match.index > lastIndex) {
+                  segments.push([false, str.substring(lastIndex, match.index)]);
+                }
+                segments.push([true, match[0]]);
+                lastIndex = match.index + match[0].length;
+              }
+              if (lastIndex < str.length) {
+                segments.push([false, str.substring(lastIndex)]);
+              }
+              return segments;
+            }
+
+            const textWrapPRdescription = (text, maxLineLength) =>
+              getSegments(/(?:^ {0,3}(```|~~~)[^]*?^ {0,3}\1 *$)|^(?:\n {4,}.*)+/gm, text)
+                .map( ([isCodeBlock, segment]) =>
+                  isCodeBlock
+                  ? segment // Don't touch code blocks
+                  : segment.split('\n').map(line =>
+                      line.length <= maxLineLength
+                      ? line // Line isn't too long, nothing to do
+                      : getSegments(/(`+).*?\1/g, line)
+                          // Treat codespans as single words:
+                          .flatMap( ([isCodeSpan, segment]) => isCodeSpan ? segment : segment.split(' ') )
+                          .reduce(([newLine, lineLength], word) =>
+                            newLine
+                            // Don't touch trailing spaces:
+                            ? word.length > 0 && lineLength + 1 + word.length > maxLineLength
+                              ? [newLine + '\n' + word, word.length]
+                              : [newLine + ' ' + word, lineLength + 1 + word.length]
+                            : [newLine + word, lineLength + word.length]
```

Comment:
This is horrible code, sorry.

Use `filter` to remove the excessive ternary. Performance doesn't matter much over here.

Reply:
Lol, yeah I figured nested ternaries would be kinda hard to read. Blame it on js for not having if-expressions :p
Started out with just for loops and mostly imperative code and slowly converted it into this.
I don't see how you would replace these with filter though, maybe I'm missing something.

Reply:
Oh true, sorry.

You can write those as like actual functions instead of chaining expressions :P

Reply:
I like the ternaries, the shape of the code fits the logic of it. Bouncing up and down between function definitions and the logic doesn't seem like at all.

Reply:
I usually put the `?` and `:` as trailing on the line previous, so the reader knows they're entering into a ternary before leaving the current line. Helps reading flow, IMO.

Reply:
Because of the nesting in this case, them trailing the previous line looks a bit confusing IMO:
```js
newLine ?
  word.length > 0 && lineLength + 1 + word.length > maxLineLength ?
    [newLine + '\n' + word, word.length] :
    [newLine + ' ' + word, lineLength + 1 + word.length] :
  [newLine + word, lineLength + word.length]
```